openwisp_monitoring/notifications/tests/test_notifications.py

```diff
@@ -292,12 +295,67 @@ def test_object_metric_multiple_notifications_no_org(self):
         self.assertIsNotNone(staff.notificationuser)
         om = self._create_object_metric(name='logins', content_object=user)
         t = self._create_threshold(metric=om, operator='>', value=90, seconds=0)
-        om._notify_users(level='info', verb='test', threshold=t)
+        om._notify_users(notification_type='default', threshold=t)
         self.assertEqual(Notification.objects.count(), 1)
         n = notification_queryset.first()
         self.assertEqual(n.recipient, admin)
         self.assertEqual(n.actor, om)
         self.assertEqual(n.target, user)
         self.assertEqual(n.action_object, om.threshold)
         self.assertEqual(n.level, 'info')
-        self.assertEqual(n.verb, 'test')
+        self.assertEqual(n.verb, 'default verb')
+
+    def test_notification_types(self):
+        self._create_admin()
+        m = self._create_general_metric(name='load')
+        self._create_threshold(metric=m, operator='>', value=90, seconds=0)
+
+        with self.subTest("Test notification for 'threshold crossed'"):
+            m.write(99)
+            n = notification_queryset.first()
+            self.assertEqual(n.level, 'warning')
+            self.assertEqual(n.verb, 'crossed threshold limit')
+            self.assertEqual(
+                n.email_subject,
+                '[example.com] warning - None has crossed threshold limit',
+            )
+            self.assertIn(
+                'Metric Load crossed threshold limit (greater than 90.0)', n.message
+            )
+
+        with self.subTest("Test notification for 'under threshold'"):
+            m.write(80)
+            n = notification_queryset.last()
+            self.assertEqual(n.level, 'info')
+            self.assertEqual(n.verb, 'returned within threshold limit')
+            self.assertEqual(
+                n.email_subject,
+                '[example.com] info - None has returned within threshold limit',
+            )
+            self.assertIn('Metric Load returned within threshold limit', n.message)
+
+    def test_notification_message(self):
+        self._create_admin()
+        om = self._create_object_metric(name='load')
+        self._create_threshold(metric=om, operator='>', value=90)
+        target = self._get_user()
+
+        with self.subTest("Test for metric exceeding threshold"):
+            om.write(99)
+            n = notification_queryset.first()
+            exp_message = (
+                '<p>warning : tester crossed threshold limit </p>\n'
+                '<p>Metric Load (user: tester) crossed threshold limit (greater than 90.0)\n'
+                f' Related : \n <a href="/admin/openwisp_users/user/{target.id}/change/">tester</a></p>'
```

Comment:
just checking, in the email the URLs are absolute right? Eg: `https://domain.com/admin/......`